### PR TITLE
Drop-down values when creating loan account at group level aren't available #358 Resolved

### PR DESCRIPTION
**Pull request by mayank-kgp**
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Apply the `MifosStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.
gradle

mvpImplemented

build passed

changesReview

correctedDropDown

### CODE REVIEW

**Comment by mayank-kgp** [comment]
@ishan1604  Please review it

**Comment by mayank-kgp** [comment]
@ishan1604  I have checked everything from my side.
Please have a look.  :)

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/grouploanaccount/GroupLoanAccountFragment.java [hunk 1]
```diff
@@ -218,10 +238,75 @@ public void onClick(View view) {
         return rootView;
     }
 
+    public void inflateLoanSpinner() {
```
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/grouploanaccount/GroupLoanAccountFragment.java]: @mayank-kgp Good work.

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/grouploanaccount/GroupLoanAccountPresenter.java [hunk 2]
```diff
@@ -75,7 +88,7 @@ public void loadGroupLoansAccountTemplate(int groupId, int productId) {
         mSubscription = mDataManager.getGroupLoansAccountTemplate(groupId, productId)
```
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/grouploanaccount/GroupLoanAccountPresenter.java]: Remove Subscription with CompositeSubscription. 
Look here https://github.com/openMF/android-client/blob/master/mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/syncgroupsdialog/SyncGroupsDialogPresenter.java

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/grouploanaccount/GroupLoanAccountPresenter.java [hunk 3]
```diff
@@ -124,5 +139,126 @@ public void onNext(Loans loans) {
...
+                .subscribe(new Action1<AmortizationTypeOptions>() {
+                    @Override
+                    public void call(AmortizationTypeOptions amortizationTypeOptions) {
+                        amortizationNameList.add(amortizationTypeOptions.getValue());
+                    }
+                });
+        return amortizationNameList;
+    }
+
+    public List<String> fillinterestcalculationperiodSpinnerOptions(final
```
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/grouploanaccount/GroupLoanAccountPresenter.java]: Same as above.

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/grouploanaccount/GroupLoanAccountPresenter.java [hunk 4]
```diff
@@ -124,5 +139,126 @@ public void onNext(Loans loans) {
...
+                    @Override
+                    public void call(InterestCalculationPeriodType interestCalculationPeriodType) {
+                        interestCalculationPeriodNameList.add(
+                                interestCalculationPeriodType.getValue());
+                    }
+                });
+        return interestCalculationPeriodNameList;
+    }
+
+    public List<String> filltransactionProcessingStrategySpinnerOptions(final
```
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/grouploanaccount/GroupLoanAccountPresenter.java]: Same as above.

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/grouploanaccount/GroupLoanAccountPresenter.java [hunk 5]
```diff
@@ -124,5 +139,126 @@ public void onNext(Loans loans) {
...
+                                             transactionProcessingStrategyOptions) {
+                        transactionProcessingStrategyNameList.add(
+                                transactionProcessingStrategyOptions.getName());
+                    }
+                });
+        return transactionProcessingStrategyNameList;
+    }
+
+
+    public List<String> fillLoanProductSpinner(final List<LoanProducts> loanProducts) {
```
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/grouploanaccount/GroupLoanAccountPresenter.java]: Same as above

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/grouploanaccount/GroupLoanAccountPresenter.java [hunk 6]
```diff
@@ -124,5 +139,126 @@ public void onNext(Loans loans) {
...
+                .subscribe(new Action1<LoanProducts>() {
+                    @Override
+                    public void call(LoanProducts loanProducts) {
+                        loanProductsNameList.add(loanProducts.getName());
+                    }
+                });
+        return loanProductsNameList;
+    }
+
+    public List<String> filltermFrequencyTypeSpinnerOptions(final List<TermFrequencyTypeOptions>
```
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/grouploanaccount/GroupLoanAccountPresenter.java]: same as above.

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/grouploanaccount/GroupLoanAccountPresenter.java [hunk 7]
```diff
@@ -124,5 +139,126 @@ public void onNext(Loans loans) {
...
+                .subscribe(new Action1<TermFrequencyTypeOptions>() {
+                    @Override
+                    public void call(TermFrequencyTypeOptions termFrequencyTypeOptions) {
+                        termFrequencyNameList.add(termFrequencyTypeOptions.getValue());
+                    }
+                });
+        return termFrequencyNameList;
+    }
+
+    public List<String> fillloanPurposeTypeSpinnerOptions(final List<LoanPurposeOptions>
```
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/grouploanaccount/GroupLoanAccountPresenter.java]: same as above.

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/grouploanaccount/GroupLoanAccountPresenter.java [hunk 8]
```diff
@@ -124,5 +139,126 @@ public void onNext(Loans loans) {
...
+                .subscribe(new Action1<InterestTypeOptions>() {
+                    @Override
+                    public void call(InterestTypeOptions interestTypeOptions) {
+                        interestTypeNameList.add(interestTypeOptions.getValue());
+                    }
+                });
+        return interestTypeNameList;
+    }
+
+    public List<String> fillloanOfficerSpinnerOptions(final List<LoanOfficerOptions>
```
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/grouploanaccount/GroupLoanAccountPresenter.java]: same as above

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/grouploanaccount/GroupLoanAccountPresenter.java [hunk 9]
```diff
@@ -124,5 +139,126 @@ public void onNext(Loans loans) {
...
+                .subscribe(new Action1<LoanOfficerOptions>() {
+                    @Override
+                    public void call(LoanOfficerOptions loanOfficerOptions) {
+                        loanOfficerNameList.add(loanOfficerOptions.getDisplayName());
+                    }
+                });
+        return loanOfficerNameList;
+    }
+
+    public List<String> fillfundSpinnerOptions(final List<FundOptions> fundOptions) {
```
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/grouploanaccount/GroupLoanAccountPresenter.java]: same as above

**Comment by therajanmaurya** [comment]
@mayank-kgp Looks good, Please make above changes and let @ishan1604 know for final look.

**Comment by mayank-kgp** [comment]
@therajanmaurya  I have done suggested changes. Please have a final look.

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/grouploanaccount/GroupLoanAccountFragment.java [hunk 10]
```diff
@@ -27,98 +26,97 @@
...
 
     @BindView(R.id.bt_loan_submit)
-    Button bt_loan_submit;
+    Button btLoanSubmit;
 
     @Inject
     GroupLoanAccountPresenter mGroupLoanAccountPresenter;
 
-    ResponseBody mResponse;
+    GroupLoanTemplate mResponse;
```
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/grouploanaccount/GroupLoanAccountFragment.java]: Remove this unused variable.

**Comment by therajanmaurya** [comment]
@mayank-kgp Looks good now. Please make two changes and let @ishan1604 know.

**Comment by mayank-kgp** [comment]
@therajanmaurya I have done the required changes.
@ishan1604  Please give it a final review.

**Pull request merged**